main.c

```diff
@@ -188,7 +194,18 @@ main(int argc, char *argv[])
 	}
 
 	/* Get the password. */
-	if (0) {
+	if (passwd_env_arg != NULL) {
+		/* We're not allowed to modify the output of getenv(). */
+		if ((passwd_env = getenv(passwd_env_arg)) == NULL) {
+			warn0("Failed to read from ${%s}", passwd_env_arg);
+			goto err1;
+		}
+
```

Comment:
Should we do any sanity testing here?  for example, if somebody accidentally points to an empty environment variable like this?

```
FOO="" ./scrypt enc --passphrase-env FOO secret.txt secret.enc
```

I'm tempted to insist that the password be at least 1 character when reading from `--passphrase-env`, just to make sure that they didn't accidentally give the wrong `$VAR` or that their system didn't default to an empty string for a missing environment variable.  But that would go against the unix / C principle of "allow people to shoot their own feet".

Reply:
I'm inclined to allow zero-character environment variables.  Any system which silently converts an undefined variable to an empty one is going to run into all sorts of issues; for example, GNU's POSIXLY_CORRECT behaviours kick in even if the variable is defined to be empty.

Also, there's probably going to be scripts which use scrypt to do things and users who say "I don't actually *want* this encrypted".